sdk/openai/Azure.AI.OpenAI/src/Custom/ChatCompletionsOptions.cs

```diff
@@ -53,38 +55,32 @@ public partial class ChatCompletionsOptions
         /// <inheritdoc cref="CompletionsOptions.User"/>
         public string User { get; set; }
 
-        internal string NonAzureModel { get; set; }
+        internal IDictionary<string, int> InternalStringKeyedTokenSelectionBiases { get; }
+
+        internal string InternalNonAzureModelName { get; set; }
+
+        internal bool? InternalShouldStreamResponse { get; set; }
 
         /// <summary> Initializes a new instance of ChatCompletionsOptions. </summary>
-        public ChatCompletionsOptions()
+        /// <param name="messages">
+        /// The collection of context messages associated with this chat completions request.
+        /// Typical usage begins with a chat message for the System role that provides instructions for
+        /// the behavior of the assistant, followed by alternating messages between the User and
+        /// Assistant roles.
+        /// </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="messages"/> is null. </exception>
+        public ChatCompletionsOptions(IEnumerable<ChatMessage> messages)
```

Comment:
I think this may boil down to a style/guideline question: do we want to enforce `messages` as a required constructor parameter on the options?

On one hand: it *is* required, and you'll get an error if you don't provide it; having it in the constructor provides compile-time enforcement of that.

On the other hand: it "feels weird" to have it stand out that way, and the code, as far as I can tell, doesn't flow as nicely.

E.g. right now, with the constructor parameter optional, I have:
```
            var requestOptions = new ChatCompletionsOptions()
            {
                Messages =
                {
                    new ChatMessage(ChatRole.System, "You are a helpful assistant."),
                    new ChatMessage(ChatRole.User, "Can you help me?"),
                    new ChatMessage(ChatRole.Assistant, "Of course! What do you need help with?"),
                    new ChatMessage(ChatRole.User, "What temperature should I bake pizza at?"),
                },
                MaxTokens = 512,
            };
```

With it required, that'd change to:
```
            var requestOptions = new ChatCompletionsOptions(new ChatMessage[]
            {
                    new ChatMessage(ChatRole.System, "You are a helpful assistant."),
                    new ChatMessage(ChatRole.User, "Can you help me?"),
                    new ChatMessage(ChatRole.Assistant, "Of course! What do you need help with?"),
                    new ChatMessage(ChatRole.User, "What temperature should I bake pizza at?"),
            })
            {
                MaxTokens = 512,
            };
```

Which requires choosing a collection type for initialization and a much less straightforward brace-initialization pattern. It also adds an extra logical deviation from the REST structure for dubious utility, though that's not a good justification on its own without more.

Is that awkwardness just in my head and part of our existing guidelines?

Reply:
Looking at the [.NET Guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html), we have the following:
> ✅ DO provide a minimal constructor that takes only the parameters required to connect to the service.

But this is technically referring to the client constructors, and here we're talking about a model class. Though I think the spirit of the guideline still applies here: Use the constructor to ensure that you have a "minimum viable" instance. 

In your example above, is your concern partially because you think that using `MaxTokens` might be particularly common? If so, I wonder if the "right" approach is to let the emitter generate the constructors with the required parameters only, and then add custom "convenience" constructors that additionally have common optional parameters.

Reply:
Setting `MaxTokens`, `Temperature`, and a few other settings is likely to be extremely common even if not outright required, yes--and that is part of the concern around the awkwardness of mixed constructor/init-setter use per the earlier.

More than that, though, is the contrast to the established "flatness" in the REST and Python usage, e.g. in OpenAI's quickstart (python/Node.js) here: https://platform.openai.com/docs/quickstart/understand-the-code -- there, there's no structural delineation made between required and optional fields; it's just a big, flat property bag.

The advantage I see of promoting functionally required request payload members to the options constructor is the forced clarity of making viable request payloads; here, I think we don't gain much from that and that the cognitive burden of having to define common request fields (some "required," some not) in two different places outweighs the gain.

A somewhat more contemporary compromise might be init-only setters, though I'm not sure where guidelines land on that (they call out get/set semantics but I don't see any mention of init).

Would you object to keeping things "flat" for the time being and noting this pattern as a discussion topic for the next round of holistic surface review?

Reply:
Sounds good. 👍